### ISSUE TITLE
rsyslog 8.32 writes my app logs to syslog even after & stop

### ISSUE THREAD

**Issue by aleeamini** [rsyslog/rsyslog#4972]
### Expected behavior
I have an application that produces logs to ```local1, local2 and local3```. I wrote a ```10-myapp.conf ``` and separated my logs to another files.
Also I append ```& stop``` to end of each line of my config file to stop logging to ```/var/log/syslog```.

### Actual behavior
I restart rsyslog service, and my app's logs generated But rsyslog continue writing my app's logs  also to ```/var/log/syslog``` file.

### Environment
- rsyslog version: 8.32.0
- platform: Ubuntu 18.04.6 LTS Bionic
## - /etc/rsyslog.conf:

```
#  /etc/rsyslog.conf    Configuration file for rsyslog.
#
#                       For more information see
#                       /usr/share/doc/rsyslog-doc/html/rsyslog_conf.html
#
#  Default logging rules can be found in /etc/rsyslog.d/50-default.conf

#################
#### MODULES ####
#################

module(load="imuxsock") # provides support for local system logging
#module(load="immark")  # provides --MARK-- message capability

# provides UDP syslog reception
#module(load="imudp")
#input(type="imudp" port="514")

# provides TCP syslog reception
#module(load="imtcp")
#input(type="imtcp" port="514")

# provides kernel logging support and enable non-kernel klog messages
module(load="imklog" permitnonkernelfacility="on")

###########################
#### GLOBAL DIRECTIVES ####
###########################

#
# Use traditional timestamp format.
# To enable high precision timestamps, comment out the following line.
#
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat

# Filter duplicated messages
$RepeatedMsgReduction on

#
# Set the default permissions for all log files.
#
$FileOwner syslog
$FileGroup adm
$FileCreateMode 0640
$DirCreateMode 0755
$Umask 0022
$PrivDropToUser syslog
$PrivDropToGroup syslog

#
# Where to place spool and state files
#
$WorkDirectory /var/spool/rsyslog

#
# Include all config files in /etc/rsyslog.d/
#
$IncludeConfig /etc/rsyslog.d/*.conf
```
## - /etc/rsyslog.d/10-myapp.conf:
```
local1.*        /var/log/myapp/app.log
local1.*        @192.168.1.20:1514
& stop
local2.*        /var/log/myapp/app_messages.log
local2.*        @192.168.1.20:1515
& stop
local3.*        /var/log/myapp/app_debug.log
local3.*        @192.168.1.20:1516
& stop

```
## - /etc/rsyslog.d/50-default.conf:
```
#  Default rules for rsyslog.
#
#                       For more information see rsyslog.conf(5) and /etc/rsyslog.conf

#
# First some standard log files.  Log by facility.
#
auth,authpriv.*                 /var/log/auth.log
*.*;auth,authpriv.none         -/var/log/syslog
#cron.*                         /var/log/cron.log
#daemon.*                       -/var/log/daemon.log
kern.*                          -/var/log/kern.log
#lpr.*                          -/var/log/lpr.log
mail.*                          -/var/log/mail.log
#user.*                         -/var/log/user.log

#
# Logging for the mail system.  Split it up so that
# it is easy to write scripts to parse these files.
#
#mail.info                      -/var/log/mail.info
#mail.warn                      -/var/log/mail.warn
mail.err                        /var/log/mail.err

#
# Some "catch-all" log files.
#
#*.=debug;\
#       auth,authpriv.none;\
#       news.none;mail.none     -/var/log/debug
#*.=info;*.=notice;*.=warn;\
#       auth,authpriv.none;\
#       cron,daemon.none;\
#       mail,news.none          -/var/log/messages

#
# Emergencies are sent to everybody logged in.
#
*.emerg                         :omusrmsg:*

#
# I like to have messages displayed on the console, but only on a virtual
# console I usually leave idle.
#
#daemon,mail.*;\
#       news.=crit;news.=err;news.=notice;\
#       *.=debug;*.=info;\
#       *.=notice;*.=warn       /dev/tty8
```

**Comment by Yousha** [comment]
Hi, I have same problem...
Any solution/fix?

**Comment by rgerhards** [comment]
I guess there are some error messages from rsyslog regarding the config. Please check if you see anything in the message log.

Or, maybe, this is a version which reserved the include order. Note that 8.32 is ancient, so this may be the case. Simple to try by renaming 10-... to 100-...  Chance is slim, but worth a try as effort-less ;-)

**Comment by aleeamini** [comment]
> I guess there are some error messages from rsyslog regarding the config. Please check if you see anything in the message log.
> 
> Or, maybe, this is a version which reserved the include order. Note that 8.32 is ancient, so this may be the case. Simple to try by renaming 10-... to 100-... Chance is slim, but worth a try as effort-less ;-)


No there is no error from rsyslog. I also ran ```rsyslogd -d``` to check if there is any error.
I also changed my config file to 60-...,100-..., 01-... and nothing changed.
Also I can't update rsyslog service because the target OS is a LTS ubuntu and there is no internet connection.
Please help me.

**Comment by rgerhards** [comment]
A slight chance to enlighten the situation via debug log:

Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

*Note: I am not sure if the instructions work for that old version. But chances are good.*

Be sure that the problem in question occurs at least once during the debug run. Once this is done, please post the debug log (e.g. via pastbin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

**Comment by aleeamini** [comment]
## The problem solved.
My application runs as a service and I forgot to turn off logging in systemd.
I edit my service systemd configurtion and the problem solved.  
in the ```/etc/systemd/system/myapp.service``` set ```StandardOutput``` and ```StandardError``` values to ```null``` to stop logging the output to syslog:
```
[Service]
Type=simple
ExecStart=/usr/bin/myapp
Restart=always
RestartSec=5
StandardOutput=null
StandardError=null
SyslogIdentifier=%n

```

**Comment by aleeamini** [comment]
> A slight chance to enlighten the situation via debug log:
> 
> Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html
> 
> _Note: I am not sure if the instructions work for that old version. But chances are good._
> 
> Be sure that the problem in question occurs at least once during the debug run. Once this is done, please post the debug log (e.g. via pastbin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

## The problem solved.
My application runs as a service and I forgot to turn off logging in systemd.
I edit my service systemd configurtion and the problem solved.  
in the ```/etc/systemd/system/myapp.service``` set ```StandardOutput``` and ```StandardError``` values to ```null``` to stop logging the output to syslog:
```
[Service]
Type=simple
ExecStart=/usr/bin/myapp
Restart=always
RestartSec=5
StandardOutput=null
StandardError=null
SyslogIdentifier=%n

```

Thanks for your attention.
